base/dsfs/dataset.go

```diff
@@ -38,6 +38,16 @@ var BodySizeSmallEnoughToDiff = 20000000 // 20M or less is small
 // LoadDataset reads a dataset from a cafs and dereferences structure, transform, and commitMsg if they exist,
 // returning a fully-hydrated dataset
 func LoadDataset(ctx context.Context, store cafs.Filestore, path string) (*dataset.Dataset, error) {
+	// set a timeout to handle long-lived requests when connected to IPFS.
+	// if we don't have the dataset locally, IPFS will reach out onto the d.web to
+	// attempt to resolve previous hashes. capping the duration  to 700 MS yeilds
+	// quick results.
+	// TODO (b5) - The proper way to solve this is to feed a local-only IPFS store
+	// to this entire function, or have a mechanism for specifying that a fetch
+	// must be local
+	ctx, cancel := context.WithTimeout(ctx, time.Millisecond*700)
```

Comment:
Thinking about these timeouts, I'd actually prefer having them in the `config` this way if anybody has issues (specific to their setup/network etc) they can actually tune this (or -1 to turn off entirely?)

Reply:
Love that you've identified what we're doing here: request timeouts. I'll unify this value to a variable in this PR & we can make it a config value in the future. adding & removing from config is a little precarious this close to a release